### DISCUSSION TITLE
RabbitMQ: Restarted erlang machine.

### DISCUSSION THREAD

**Discussion by mrodriguez-aviles** [rabbitmq/rabbitmq-server#3840] [Questions] [answered]
OS: Server 2019
Setup - 3 server cluster with a rabbitMQ installed on each server.

issue: noticed that the node uptime on the management gui will so that the node was only up for a few hours, but the server has not been restarted.  Looking at the event viewer logs, I found this  "RabbitMQ: Restarted erlang machine."

RabbitMQ version: 3.9.8
ERlang version: 24

Additionally, this is a fresh install of the RabbitMQ with these newer servers.  Did not experience this issue with the previous ones, additionally do not seem to be experiencing this with my test server (which sits in QA and is a single server and not in a cluster).

I have attached a couple of screenshots to hopefully share what I am seeing.

I'm hoping that somebody could point me in the right direction to resolve the issue as Google has failed me.

![erlang_event_viewer](https://user-images.githubusercontent.com/58010854/144521340-69e3e31f-3a0d-488e-922d-a8f686084cc4.png)
![dashboard_with_uptime](https://user-images.githubusercontent.com/58010854/144521454-64e43441-12de-4844-90c6-c15e6efc4b2e.png)

**Comment by lukebakken** [comment, marked as answer]
```
lbakken@shostakovich ~/development/erlang/otp (master=)
$ git grep -i 'restarted erlang'
erts/etc/win32/erlsrv/erlsrv_service.c:911:     log_warning(L"Restarted erlang machine.");
```

This comes from the program that runs the Erlang VM (and thus RabbitMQ) as a Windows service:

https://github.com/erlang/otp/blob/master/erts/etc/win32/erlsrv/erlsrv_service.c#L911

It appears that RabbitMQ/Erlang on your server is in a restart loop. I've never seen this. You may get some answers in the log files at `%AppData%\RabbitMQ\log`. If you'd like to compress them and attach them here I'll take a look.

All I can suggest is uninstall RabbitMQ and Erlang, and re-install using the **latest versions** of each - RabbitMQ 3.9.11 and Erlang 24.1.7

https://erlang.org/download/otp_versions_tree.html

**Reply by mrodriguez-aviles** [reply]
Thank you!  Upgrading the version of Erlang helped resolve the issue.
